### PR TITLE
Use only jdbi in async job runner

### PR DESCRIPTION
**Pull request by jaahon**
#### Summary

Use only jdbi in async job runner and the functions that handle the async jobs.

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt [hunk 1]
```diff
@@ -90,30 +93,23 @@ class DecisionService(
...
 
-    fun updateDecisionGuardianDocumentUri(decisionId: UUID, pdfUri: URI) {
-        withSpringHandle(dataSource, { h -> updateDecisionGuardianDocumentUri(h, decisionId, pdfUri) })
-    }
-
-    fun updateDecisionOtherGuardianDocumentUri(decisionId: UUID, pdfUri: URI) {
-        withSpringHandle(dataSource, { h -> updateDecisionOtherGuardianDocumentUri(h, decisionId, pdfUri) })
-    }
-
     @Transactional
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt:96]: `@Transactional` + `h: Handle` = :bomb:

File: service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt [hunk 2]
```diff
@@ -92,53 +91,43 @@ class DecisionGenerator(
     }
 
     @Transactional
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt:93]: `@Transactional` + `h: Handle` = :bomb:

File: service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt [hunk 3]
```diff
@@ -92,53 +91,43 @@ class DecisionGenerator(
...
 
-        withSpringHandle(dataSource) { h ->
-            val families = findFamiliesByChild(h, childId, period)
-            handleDecisionChangesForFamilies(h, period, families)
-        }
+        val families = findFamiliesByChild(h, childId, period)
+        handleDecisionChangesForFamilies(h, period, families)
     }
 
     @Transactional
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt:101]: `@Transactional` + `h: Handle` = :bomb:

File: service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt [hunk 4]
```diff
@@ -92,53 +91,43 @@ class DecisionGenerator(
...
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt:109]: `@Transactional` + `h: Handle` = :bomb:

File: service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt [hunk 5]
```diff
@@ -92,53 +91,43 @@ class DecisionGenerator(
...
 
-        withSpringHandle(dataSource) { h ->
-            val families = findFamiliesByHeadOfFamily(h, adultId, period) + findFamiliesByPartner(h, adultId, period)
-            handleDecisionChangesForFamilies(h, period, families)
-        }
+        val families = findFamiliesByHeadOfFamily(h, adultId, period) + findFamiliesByPartner(h, adultId, period)
+        handleDecisionChangesForFamilies(h, period, families)
     }
 
     @Transactional
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt:117]: `@Transactional` + `h: Handle` = :bomb:

File: service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt [hunk 6]
```diff
@@ -92,53 +91,43 @@ class DecisionGenerator(
...
 
-        withSpringHandle(dataSource) { h ->
-            val families = findFamiliesByHeadOfFamily(h, personId, period) + findFamiliesByPartner(h, personId, period)
-            handleDecisionChangesForFamilies(h, period, families)
-        }
+        val families = findFamiliesByHeadOfFamily(h, personId, period) + findFamiliesByPartner(h, personId, period)
+        handleDecisionChangesForFamilies(h, period, families)
     }
 
     @Transactional
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DecisionGenerator.kt:125]: `@Transactional` + `h: Handle` = :bomb:

File: service/src/main/kotlin/fi/espoo/evaka/pis/service/FamilyInitializerService.kt [hunk 7]
```diff
@@ -42,11 +42,11 @@ class FamilyInitializerService(
     }
 
     @Transactional
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/pis/service/FamilyInitializerService.kt:44]: `@Transactional` + `h: Handle` = :bomb:

File: service/src/main/kotlin/fi/espoo/evaka/pis/service/MergeService.kt [hunk 8]
```diff
@@ -97,21 +98,24 @@ class MergeService(
...
+            h.createQuery(parentsSQL)
+                .bind("id", master)
+                .map { rs, _ -> rs.getUUID("head_of_child") }
+                .forEach { parentId ->
+                    sendFamilyUpdatedMessage(h, parentId, feeAffectingDateRange.first, feeAffectingDateRange.second)
+                }
         }
     }
 
     @Transactional
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/pis/service/MergeService.kt:110]: `@Transactional` + `h: Handle` = :bomb:

**Comment by jaahon** [comment]
Good point 😬

**Review by Gekkio — APPROVED** [review]

**Pull request merged**